### PR TITLE
Revert "Scroll to joined/activated channel"

### PR DESCRIPTION
**Pull request by dgw**
Reverts #2166

Since merging, this has caused too many issues due to inconsistent browser support (#2233, for example).

I'm working on a more brute-force approach that doesn't use this flaky API. Until that's done, I'd argue that removing the functionality is better than the current state.

### CODE REVIEW

**Comment by astorije** [comment]
That's too bad :(

I don't know if that's any useful to you, but [this is what I originally did for the userlist scrolling-into-view](https://github.com/thelounge/thelounge/pull/1842/commits/ee7272305af0a6f41ccaa86f050896e6d40fb49c#diff-0286bab841c266a97e65f3863daf8488R86).

**Comment by astorije** [comment]
Alternatively we could try one of the scrollIntoView polyfills?

**Comment by xPaw** [comment]
I don't see a reason to revert this, we can just fix the function.

As for a polyfill, something like this? https://github.com/stipsan/scroll-into-view-if-needed

**Comment by dgw** [comment]
@astorije Your code didn't get me very far the last time I looked at it, because the user list is all one big container but channels are nested in multiple parent elements relative to the container we care about scrolling, so it's trickier to do the same thing in a generic way. I might look at it again next week, or I might not, depends on how we get on with polyfills. That was actually my first idea, but I wasn't too keen on adding Yet Another Dependency if it could be done relatively cheaply in a bespoke utility function.

Basically, I see this revert probably isn't going to happen. I will come back to this, but likely not until Monday at the earliest. My weekend is very, very busy in the offline world, so I won't be coding much. :sweat_smile:

**Comment by McInkay** [comment]
We don't need to rush @dgw , this code is not causing a massive issue and is not in a release, so it's not like it needs immediately fixing.

**Comment by astorije** [comment]
> As for a polyfill, something like this? stipsan/scroll-into-view-if-needed

The risk with polyfills that apply if needed is that Safari _kinda_ supports it but not really, so I don't know if polyfills detect that scroll into view is available and just stop there.

https://github.com/iamdustan/smoothscroll seems to have a `scrollIntoView` polyfill, and a way to force it... Could this be a potential candidate?

**Comment by xPaw** [comment]
> The risk with polyfills that apply if needed is that Safari kinda supports it but not really

Polyfill wasn't the right word to use it, as that library has a separate implementation and function export. It doesn't try to use the `Element.scrollIntoView` function. So we should be safe there as far as cross browser support is concerned.

**Comment by astorije** [comment]
Aaah, I see, it's not "if needed" as in "this polyfill applies when native support is not available", it's "if needed" as in "scroll the element if it's not already in view", gotcha.

Yeah, trying with the project you suggested would be nice.

@dgw, I'm going to close this PR because we all agreed we prefer to try an alternative, the issue is bearable since we're not releasing stable right away (and it's already in a pre-release), there are conflicts in this PR now and I'm sure you don't want to deal with conflicts on an automated revert commit/PR :D

**Comment by dgw** [comment]
That's fine, @astorije, you're right of course about the conflicts :sweat_smile:

Still planning to try and address this next week. My marathon weekend is nearly over :tada:
